tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -66,9 +66,14 @@ static const char * const devices[] = {
 	LABELS_FOR_DT_COMPAT(microchip_xec_timer)
 	LABELS_FOR_DT_COMPAT(nxp_imx_epit)
 	LABELS_FOR_DT_COMPAT(nxp_imx_gpt)
+#ifdef CONFIG_COUNTER_MCUX_RTC
 	LABELS_FOR_DT_COMPAT(nxp_kinetis_rtc)
+#endif
 	LABELS_FOR_DT_COMPAT(silabs_gecko_rtcc)
 	LABELS_FOR_DT_COMPAT(st_stm32_rtc)
+#ifdef CONFIG_COUNTER_MCUX_PIT
+	LABELS_FOR_DT_COMPAT(nxp_kinetis_pit)
+#endif
```

Comment:
Neither of these ifdefs should be necessary

Reply:
the LABELS_FOR_DT_COMPAT can't tell whether the counter driver for rtc or pit is enable. it only checks whether the RTC0/PIT0 exist or not. and I find if I enable RTC0 in counter test, it fails, I need report one jira issue for this, as it is not default enabled. so we miss this test before.

Reply:
Ok, please create a github issue and copy a link here.

Reply:
#26832 reported for frdm_k82f RTC counter